src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/conf/WriteYamlStep.java

```diff
@@ -121,25 +133,32 @@ public String getFunctionName() {
         }
 
         @Override
+        @Nonnull
         public String getDisplayName() {
             return "Write a yaml from an object.";
         }
     }
 
-    public static class Execution extends AbstractSynchronousNonBlockingStepExecution<Void> {
+    public static class Execution extends SynchronousNonBlockingStepExecution<Void> {
         private static final long serialVersionUID = 1L;
 
-        @StepContextParameter
         private transient TaskListener listener;
 
-        @StepContextParameter
         private transient FilePath ws;
```

Comment:
Recommend deleting fields like this and just look up as local variables on demand.